lib/blockchain_double.js

```diff
@@ -418,29 +418,35 @@ BlockchainDouble.prototype.sortByPriceAndNonce = function() {
   self.pending_transactions = sorted_transactions;
 };
 
-BlockchainDouble.prototype.processCall = function(tx, callback) {
+BlockchainDouble.prototype.processCall = function(tx, blockNumber, callback) {
   var self = this;
+  var startingStateRoot = self.stateTrie.root;
```

Comment:
I wouldn't address this now, but the fact that we have this root updating nonsense slung all over the place is a bit of of a code smell, suggesting that we're really not encapsulating this concern very well.

I have a branch that creates a new instance of the trie and EVM on demand with the correct root, but the way it's structured really isn't much better. If you feel like taking this on, let me know. Even if you don't, it'd be good to hash out a plan of attack with someone else on zoom if you're willing.

Reply:
Yes!